llama-index-integrations/memory/llama-index-memory-mem0/llama_index/memory/mem0/base.py

```diff
@@ -0,0 +1,201 @@
+from typing import Dict, List, Optional, Union, Any
+from llama_index.core.memory.chat_memory_buffer import ChatMemoryBuffer
+from llama_index.core.memory.types import BaseMemory
+from llama_index.memory.mem0.utils import (
+    convert_memory_to_system_message,
+    convert_chat_history_to_dict,
+    convert_messages_to_string,
+)
+from mem0 import MemoryClient, Memory
+from pydantic import (
+    BaseModel,
+    ConfigDict,
+    Field,
+    ValidationError,
+    model_validator,
+    SerializeAsAny,
+    PrivateAttr,
+)
+from llama_index.core.base.llms.types import ChatMessage, MessageRole
+
+
+class BaseMem0(BaseMemory):
+    """Base class for Mem0."""
+
+    _client: Optional[Union[MemoryClient, Memory]] = PrivateAttr(default=None)
+
+    model_config = ConfigDict(arbitrary_types_allowed=True)
+
+    def __init__(self, **data) -> None:
+        super().__init__(**data)
+        if "client" in data:
+            object.__setattr__(self, "_client", data["client"])
```

Comment:
This feels a little janky

I would rewrite this

```
class BaseMem0(BaseMemory):
     """Base class for Mem0."""
     model_config = ConfigDict(arbitrary_types_allowed=True)
     
     client: Union[MemoryClient, Memory]
```

It seems like client is a required arg right?
```
memory = BaseMem0(client=client)
```